src/Processors/Port.h

```diff
@@ -214,12 +214,16 @@ class Port
 public:
     using Data = State::Data;
 
-    Port(Block header_) : header(std::move(header_)) {} /// NOLINT
-    Port(Block header_, IProcessor * processor_) : header(std::move(header_)), processor(processor_) {}
+
+    Port(ConstBlockPtr header_) : header(std::move(header_)) { } // NOLINT(google-explicit-constructor)
+    Port(Block && header_) : header(std::make_shared<const Block>(std::move(header_))) { } // NOLINT(google-explicit-constructor)
+    Port(const Block & header_) : header(std::make_shared<const Block>(header_)) { } // NOLINT(google-explicit-constructor)
+    Port(Block header_, IProcessor * processor_) : header(std::make_shared<const Block>(std::move(header_))), processor(processor_) { }
```

Comment:
I assume we want conversion happening without explicitly calling the ctor. As the previously existing ctor also used in such way, I think it is okay to merge this.

Reply:
Yes, I have attempted to add the explicit keyword, but it resulted in numerous compilation errors.